### PR TITLE
refactor(plugins): Stateless historical plugin

### PR DESCRIPTION
**Pull request by oofncubed**

### CODE REVIEW

File: eth/core/host.go [hunk 1]
```diff
@@ -148,20 +150,20 @@ type (
 		// HistoricalPlugin implements `libtypes.Preparable`.
 		libtypes.Preparable
```
Comment (calbera) [inline eth/core/host.go:151]: you can now remove this. basically .Prepare should never be called on the historical plugin

File: cosmos/x/evm/plugins/historical/historical_data.go [hunk 2]
```diff
@@ -38,12 +39,13 @@ import (
 // TODO: WHO WROTE THIS CODE THE FIRST TIME BLS FIX IT IS HORRIBLE.
```
Comment (calbera) [inline cosmos/x/evm/plugins/historical/historical_data.go:39]: you can remove this line bls bls

**Comment by calbera** [comment]
make this historical plugin only. Separate PRs for config, gas, and maybe block

**Comment by itsdevbear** [comment]
Is this good?

**Comment by oofncubed** [comment]
> Is this good?

no, the read paths won't work. the context passed into the Get functions won't be able to unwrap into a sdk.context

**Comment by itsdevbear** [comment]
lets shelve this for now tbh.

**Pull request closed without merging**